feat(health): support show health in floating window (#31086)
neovim/neovim: runtime/doc/news.txt

• |vim.diagnostic.setqflist()| updates an existing quickfix list with the
  given title if found
• |ui-messages| content chunks now also contain the highlight group ID.

> • |health| can be displayed in a floating window and controlled by

Reviewer: ```suggestion
• |:checkhealth| can display in a floating window, controlled by
```
